### PR TITLE
[WOR-757] Fail Azure workspace/cloud context deletion if there are controlled resources that user cannot delete

### PR DESCRIPTION
**Pull request by marctalbott**
Ticket: [WOR-757](https://broadworkbench.atlassian.net/browse/WOR-757)
- Users cannot delete controlled application resources in their workspace. We need to check that a user can actually delete all controlled resources in their workspace _before_ we start deleting the cloud objects, not after.

[WOR-757]: https://broadworkbench.atlassian.net/browse/WOR-757?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by blakery — APPROVED** [review]

File: service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteControlledAzureResourcesStepTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import bio.terra.workspace.service.resource.controlled.model.ManagedByType;
+import bio.terra.workspace.service.resource.model.CloningInstructions;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+public class DeleteControlledAzureResourcesStepTest extends BaseAzureUnitTest {
```
Comment (cahrens) [inline service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteControlledAzureResourcesStepTest.java:33]: Should we also add a test of the happy path? There is some coverage in `deleteMcWorkspaceWithAzureContextAndResource`, but it seems odd that this file didn't already have a unit test.

**Review by cahrens — APPROVED** [review]
Consider adding a happy path test as well.

File: service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteControlledAzureResourcesStepTest.java (hunk 1 above)
Comment (marctalbott) [reply service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteControlledAzureResourcesStepTest.java:33]: I was on the fence about this, and this is enough to push me over. I'll add it

Comment (cahrens) [reply service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteControlledAzureResourcesStepTest.java]: Thanks!

**Pull request merged**